### PR TITLE
Initial commit for aws s3 remove full access to authenticated users

### PR DESCRIPTION
**Pull request by sreedevikr**
The commit includes the code changes for removing the full access privileges for aws authenticated users. Please provide us more information about the rule id , we have not updated the rule id for this remediation.

### CODE REVIEW

**Review by Manmohan-Tyagi — DISMISSED** [review]

File: remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Remove S3 Public Admin ACL
+
+This job will remove the public "full access privileges" permission. All other ACL permissions will be left alone. This means that if "AuthenticatedUsers" have access to FULL_CONTROL, the permissions will be changed to remove the 'FULL_CONTROL' permission. 
+
+### Applicable Rule
+
+##### Rule ID:
+5c8c26617a550e1fb6560c69
```
Comment (kshrutik) [inline remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/README.md:8]: Rule Id for s3 full access for authenticated users is this - 5c8c26567a550e1fb6560c5d

File: remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+##### Rule Name:
+The S3 bucket policy allows unrestricted access
```
Comment (kshrutik) [inline remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/README.md:11]: "S3 bucket should not give full access to all authenticated users" - this is the rule name.

File: remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Getting Started
+
+### Prerequisites
+
+The provided AWS credential must have access to `s3:PutBucketPublicAccessBlock`.
```
Comment (kshrutik) [inline remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/README.md:17]: The provided AWS credential must have access to `s3:GetBucketAcl` and `s3:PutBucketAcl`

**Review by kshrutik — CHANGES_REQUESTED** [review]

**Comment by kshrutik** [comment]
Please add unit test

**Comment by kshrutik** [comment]
Also, can you please update the main readme file and the tox file? Run the tox file and verify if the unit tests are passing.
Refer this [PR](https://github.com/vmware-samples/secure-state-remediation-jobs/pull/97)

File: remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/aws_s3_remove_fullaccess_authenticatedusers.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        """Block public access ACL to authenticated users
+
+        :param client: Instance of the AWS boto3 client.
+        :param bucket_name: The name of the bucket for which to block access.
+        :type bucket_name: str.
+        :returns: Integer signaling success or failure
+        :rtype: int
+        :raises: botocore.exceptions.ClientError
+        """
+
```
[the remark this answers is not in the document]
Comment (sreedevikr) [reply remediation_worker/jobs/aws_s3_remove_fullaccess_authenticatedusers/aws_s3_remove_fullaccess_authenticatedusers.py:60]: @kshrutik Thanks for the review comments  .  I have included the fix for the review comments. Could you please review .

**Comment by sreedevikr** [comment]
@kshrutik Thanks for the review comments.Updated the main readme and the tox file

**Review by kshrutik — APPROVED** [review]

**Pull request merged**